setup.cfg

```diff
@@ -0,0 +1,192 @@
+    GitHub: repo = https://github.com/deepset-ai/haystack
+description = Neural Question Answering & Semantic Search at Scale. Use modern transformer based models like BERT to find answers in large document collections
+long_description = file: README.md
+long_description_content_type = text/markdown
+keywords=
+    QA 
+    Question-Answering 
+    Reader 
+    Retriever 
+    semantic-search 
+    search 
+    BERT 
+    roberta 
+    albert 
+    squad 
+    mrc 
+    transfer-learning 
+    language-model 
+    transformer
+author = Malte Pietsch, Timo Moeller, Branden Chan, Tanay Soni
+author_email = malte.pietsch@deepset.ai
+license = Apache License 2.0
+license_file = LICENSE
+platforms = any
+classifiers =
+    Development Status :: 5 - Production/Stable
+    Intended Audience :: Science/Research
+    License :: Freely Distributable
+    License :: OSI Approved :: Apache Software License
+    Topic :: Scientific/Engineering :: Artificial Intelligence
+    Operating System :: OS Independent
+    Programming Language :: Python
+    Programming Language :: Python :: 3
+    Programming Language :: Python :: 3.7
+    Programming Language :: Python :: 3.8
+    Programming Language :: Python :: 3.9
+    Programming Language :: Python :: 3.10
+
+[options]
+use_scm_version = True
+python_requires = >=3.7
+packages = find:
+include_package_data = True
+test_suite = tests
+setup_requires =
+    setuptools >= 46.4.0
+install_requires =
+    importlib-metadata; python_version < '3.8'
+    torch>1.9,<1.11  # pytorch
+    tqdm  # progress bars in model download and training scripts
+    requests  # Used for downloading models over HTTP
+    pydantic  # Validation of the core dataclasses (Document, Label, etc...)
+    scipy>=1.3.2  # for stats in run_classifier
+    scikit-learn>=1.0.0  # for stats in run_classifier
+    seqeval  # Metrics or logging related
+    mlflow<=1.13.1  # Metrics or logging related
+    transformers==4.13.0  # huggingface transformers
+    dill  # pickle extension for (de-)serialization
+    #onnxruntime  # Inference with ONNX models. Install onnxruntime-gpu for Inference on GPUs
+    #onnxruntime_tools  # Inference with ONNX models. Install onnxruntime-gpu for Inference on GPUs
+    psutil
+    pandas
+    langdetect   # for PDF conversions
+    pytesseract==0.3.7  # for PDF conversions using OCR
+    pillow  # for PDF conversions using OCR
+    pdf2image==1.14.0  # for PDF conversions using OCR
+    sentence-transformers>=0.4.0
+    python-multipart
+    python-docx  # To read DOCX files
+    tika
+    uvloop==0.14; sys_platform != 'win32' and sys_platform != 'cygwin'
+    httptools
+    nltk
+    more_itertools
+    networkx
+    mmh3
+    dataclasses-json
+    quantulum3
+    azure-ai-formrecognizer==3.2.0b2
+
+
+[options.packages.find]
+exclude =
+    rest_api*
+    test*
+    tutorials*
+    ui*
+
+
+[options.extras_require]
+elasticsearch=
+    elasticsearch>=7.7,<=7.10
+    elastic-apm
```

Comment:
I would make elasticsearch part of the default installation. It's probably the most-used document store + many of our tutorials use it + the dependencies are very lightweight and never made trouble in the past.

Reply:
I'm not strongly against it, but before proceeding I want to highlight that minimal == default in Python's packaging world. So whatever we add to the default installation becomes mandatory for every other setup. That's why I put also Elasticsearch in its own group: to allow, for example, Milvus user to not install ES. I'd rather define a `default` extra group and add ES deps there. Then we should promote the use of this extra a bit by using it always in tutorials and examples, but I see it doable.

However, if it looks too cumbersome I can also add ES to the minimal install. It can be easily changed later, although it would count as a breaking change.

Reply:
I get your point, but if we take the "minimal == default" direction to the extreme, we'd also need to exclude things like sentence transformers and many others as they are only used in a few nodes. 
I'd rather suggest we go into the direction of "most common == default". What do we want to achieve with this PR? It's about improving the developer experience and ensuring a seamless, fast installation without any frustration. Moving annoying deps like faiss to "optional" helps a lot. However, if the average user has to learn first about the right installation command (and our definition of different bundles) or encounters many of our new "ImportErrors" in the first hours of usage, it's going in the wrong direction of dev experience.

Reply:
Sounds good: I will add back ES deps to the `install_requires` group. However, if you agree I'll keep the import checks as they are right now: this can help expert users to workaround the mandatory ES dependencies, in case they need to. Is that ok?